Formula/haskell-stack.rb

```diff
@@ -19,14 +42,34 @@ class HaskellStack < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "1c0e6d39df1e8c28c0ed815df4a2f02a3e302a758fb9dade1aaf3d13212ce5ad"
   end
 
+  head do
+    url "https://github.com/commercialhaskell/stack.git", branch: "master"
+
+    # Due to aeson-2.0.0.0 update, stack can no longer be built with cabal.
+    # Dependencies freezing is required to build stack with cabal.
+    #
+    # Reference: https://github.com/commercialhaskell/stack/pull/5677
+    patch do
+      url "https://github.com/commercialhaskell/stack/commit/b2a2e56c.patch?full_index=1"
+      sha256 "43c59b4af52e372cf4af14514f1c89b62bc450171eed344aeed46e2d9d9c0d96"
+    end
+  end
```

Comment:
Let's leave `head` alone -- it's not tested, and the PR could be merged at any time.

Reply:
Sorry, by "not tested" I mean "not tested in CI", so what is and isn't the official build tool of stack isn't relevant.

That said, we've been building stack with cabal for about 7 years now, and there are about 30,000 downloads of `haskell-stack` a year, so even "way less tested" in your terms is "pretty damn thoroughly tested".

Reply:
To put my point another way: if we really must patch `head` to get a working build, then instead let's just remove it. We don't carry around patched for head builds.

Reply:
Fair point. I’ll revert the head block.